salt/modules/state.py

```diff
@@ -2133,6 +2125,46 @@ def pkg(pkg_path,
     return ret
 
 
+def _cleanup_debug_pkgs(pkg_dir):
+    '''
+    Remove previously uploaded salt-ssh state packages.
+    Keep the most recent 10 packages for debugging purposes.
+    '''
+    pkgs = []
+    for s in os.listdir(pkg_dir):
+        if not (s.startswith('debug_salt_state') and s.endswith('.tgz')):
+            continue
+        pkg_path = os.path.join(pkg_dir, s)
+        pkgs.append((pkg_path, os.stat(pkg_path).st_mtime))
+    n_to_keep = 10
+    for pkg_path, _ in sorted(pkgs, key=lambda pkg: pkg[1])[:-n_to_keep]:
+        os.unlink(pkg_path)
+
+
+def pkg(pkg_path, remove_pkg=False, *args, **kwargs):
+    '''
+    Execute a packaged state run, the packaged state run will exist in a
+    tarball available locally. This packaged state
+    can be generated using salt-ssh.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' state.pkg /tmp/salt_state.tgz 760a9353810e36f6d81416366fc426dc md5
+    '''
+    try:
+        return _pkg(pkg_path, *args, **kwargs)
+    finally:
+        if remove_pkg:
+            # Keep the copy for debugging purposes.
+            pkg_dir = os.path.dirname(pkg_path)
+            debug_pkg_path = os.path.join(
+                pkg_dir, 'debug_%s' % os.path.basename(pkg_path))
```

Comment:
please use format strings here.

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/22650/violations/file/salt/modules/state.py/

Or just `'debug_' + os.path.basename(pkg_path)` which is faster.